### PR TITLE
openapi fixes

### PR DESCRIPTION
**Pull request by nitinmohan87**
Required by https://github.com/goadesign/plugins/pull/15

### CODE REVIEW

File: http/codegen/openapi/openapi_v2_builder.go [hunk 1]
```diff
@@ -110,6 +110,35 @@ func NewV2(root *httpdesign.RootExpr) (*V2, error) {
 	return s, nil
 }
 
+// ExtensionsFromExpr generates swagger extensions from the given metadata
+// expression.
+func ExtensionsFromExpr(mdata design.MetadataExpr) map[string]interface{} {
```
Comment (raphael) [inline http/codegen/openapi/openapi_v2_builder.go]: why does this have to be a public method?

Comment (nitinmohan87) [reply http/codegen/openapi/openapi_v2_builder.go]: It is used in the security plugin

Comment (nitinmohan87) [reply http/codegen/openapi/openapi_v2_builder.go]: https://github.com/goadesign/plugins/blob/security_middleware/security/generate.go#L163

Comment (raphael) [reply http/codegen/openapi/openapi_v2_builder.go]: got it, thanks!

**Pull request merged**